src/WinRT.Runtime/ComWrappersSupport.cs

```diff
@@ -395,9 +395,10 @@ internal static string GetRuntimeClassForTypeCreation(IInspectable inspectable,
             return runtimeClassName;
         }
 
-        private static bool ShouldProvideIReference(Type type)
+        private readonly static ConcurrentDictionary<Type, bool> IsWindowsRuntimeTypeCache = new ConcurrentDictionary<Type, bool>();
+        private static bool IsWindowsRuntimeType(Type type)
         {
-            static bool IsWindowsRuntimeType(Type type)
+            static bool IsWindowsRuntimeTypeHelper(Type type)
             {
                 if ((type.GetCustomAttribute<WindowsRuntimeTypeAttribute>() is object) ||
```

Comment:
I feel like this is equivalent to Projections.IsTypeWindowsRuntimeType, is it not?

Reply:
I'd love to consolidate into a single cache, but they have different purposes and implementations.  To avoid risk of regression, we can maintain two caches for now.